### PR TITLE
lightning: migrate test-infra to testify for backend/local and reduce test time

### PR DESCRIPTION
**Pull request by gozssky**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/29980.

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: br/pkg/lightning/backend/local/local_test.go [hunk 1]
```diff
@@ -176,28 +166,28 @@ func (s *localSuite) TestRangeProperties(c *C) {
...
+	require.NoError(t, err)
 
 	// Smallest key in props.
-	c.Assert(props[0].Key, DeepEquals, cases[0].key)
+	require.Equal(t, cases[0].key, props[0].Key)
 	// Largest key in props.
-	c.Assert(props[len(props)-1].Key, DeepEquals, cases[len(cases)-1].key)
-	c.Assert(len(props), Equals, 7)
+	require.Equal(t, cases[len(cases)-1].key, props[len(props)-1].Key)
+	require.Equal(t, 7, len(props))
```
Comment (hawkingrei) [inline br/pkg/lightning/backend/local/local_test.go:175]: use ```require.Len```.

File: br/pkg/lightning/backend/local/local_test.go [hunk 2]
```diff
@@ -301,33 +291,33 @@ func (s *localSuite) TestRangePropertiesWithPebble(c *C) {
...
 	sstMetas, err := db.SSTables(pebble.WithProperties())
-	c.Assert(err, IsNil)
+	require.NoError(t, err)
 	for i, level := range sstMetas {
 		if i == 0 {
-			c.Assert(len(level), Equals, 1)
+			require.Equal(t, 1, len(level))
 		} else {
-			c.Assert(len(level), Equals, 0)
+			require.Equal(t, 0, len(level))
```
Comment (hawkingrei) [inline br/pkg/lightning/backend/local/local_test.go:312]: Ditto

File: br/pkg/lightning/backend/local/local_test.go [hunk 3]
```diff
@@ -518,19 +508,19 @@ func (s *localSuite) TestIsIngestRetryable(c *C) {
...
-	c.Assert(err, NotNil)
+	require.Equal(t, retryWrite, retryType)
+	require.Error(t, err)
 
 	resp.Error = &errorpb.Error{Message: "unknown error"}
 	retryType, _, err = local.isIngestRetryable(ctx, resp, region, metas)
-	c.Assert(retryType, Equals, retryNone)
-	c.Assert(err, ErrorMatches, "non-retryable error: unknown error")
+	require.Equal(t, retryNone, retryType)
+	require.Equal(t, "non-retryable error: unknown error", err.Error())
```
Comment (hawkingrei) [inline br/pkg/lightning/backend/local/local_test.go:523]: use ```require.EqualError```.

Comment (gozssky) [reply br/pkg/lightning/backend/local/local_test.go]: all done, thanks

**Review by hawkingrei — APPROVED** [review]

**Review by tangenta — APPROVED** [review]

**Pull request merged**